tatsu/infos.py

```diff
@@ -84,6 +92,21 @@ def new(
             result = config.replace_config(config)
         return result.replace(**settings)
 
+    # NOTE:
+    #    Using functools.cache directly makes objects of this class unhashable
+
+    @property
+    def comments_pattern(self) -> re.Pattern | None:
+        return cached_re_compile(self.comments)
+
+    @property
+    def eol_comments_pattern(self) -> re.Pattern | None:
+        return cached_re_compile(self.eol_comments)
+
+    @property
+    def whitespace_pattern(self) -> re.Pattern | None:
+        return cached_re_compile(self.whitespace)
```

Comment:
I'm not really sure we need these properties. I'd be a fan of keeping `ParserConfig` as minimal as possible and make consumers of the "final" config, after all settings have been applied, perform whatever compilation or caching they need.

As an example, the whitespace pattern property may not be used because `Buffering` already has it's own logic for how it needs to parse this, including defaulting to an alternate pattern.

Plus, making these public properties implies some degree of potential future support or analogue.

Reply:
Agreed!